src/core/tests/model.cpp

```diff
@@ -1523,3 +1523,27 @@ TEST(model, set_batch_size_validation_throw) {
         FAIL() << "Expected ov::Exception";
     }
 }
+
+TEST(model, generate_unique_names) {
```

Comment:
Please add test to check that if user set two same names they won't be overridden.